language/move-compiler/src/parser/lexer.rs

```diff
@@ -638,3 +637,72 @@ fn get_name_token(name: &str) -> Tok {
         _ => Tok::Identifier,
     }
 }
+
+// Trim the start whitespace characters, include: space, tab, lf(\n) and crlf(\r\n).
+fn trim_start_whitespace(text: &str) -> &str {
+    let mut pos = 0;
+    let mut iter = text.chars();
+
+    while let Some(chr) = iter.next() {
+        match chr {
+            ' ' | '\t' | '\n' => pos += 1,
+            '\r' => {
+                if let Some('\n') = iter.next() {
+                    pos += 2;
+                    continue;
+                }
+
+                break;
+            }
+            _ => break,
```

Comment:
This is likely fine, but you can also consider: 
```
match chr {
    ' ' | '\t' | '\n' => pos += 1,
    '\r' if matches!(iter.next(), Some('\n')) => pos += 2,
    _ => break,
};
```
But I know that while this is easier for me to read, it might not be for everyone else :)

Reply:
That's great, it's very concise.

Reply:
@tnowacki I updated the PR and passed the CI review.